spec/API_specification/linear_algebra_functions.md

```diff
@@ -255,9 +255,25 @@ Computes the outer product of two vectors `x1` and `x2`.
     -   a two-dimensional array containing the outer product and whose shape is `NxM`. The returned array must have a data type determined by {ref}`type-promotion` rules.
 
 (function-pinv)=
-### pinv()
+### pinv(x, /, *, tol=None)
 
-TODO
+Computes the (Moore-Penrose) pseudo-inverse of a matrix (or stack of square matrices) `x`.
+
+#### Parameters
+
+-   **x**: _&lt;array&gt;_
+
+    -   input array having shape `(..., M, N)` and whose innermost two dimensions form `MxN` matrices. Must have a data type of either `float32` or `float64`.
+
+-   **tol**: _Optional\[ Union\[ float, &lt;array&gt; ] ]_
+    
+    -   cutoffs for small singular values. Singular values less than or equal to `tol * largest_singular_value` are set to zero. Must be compatible with `shape(x)[:-2]` (see {ref}`broadcasting`). If a `float`, the value is equivalent to a zero-dimensional array having the same data type as `x` and is broadcast against each matrix. If `None`, the default value is `max(M, N) * eps`, where `eps` must be the floating-point epsilon associated with the data type of `x`. Default: `None`.
```

Comment:
"floating-point epsilon" -> "floating-point machine epsilon" or just "machine epsilon".

Reply:
I suppose by the time we support complex numbers in this function, the machine epsilon is the same as that of their real dtypes?

Reply:
btw out of curiosity I looked up Julia's `pinv` and found they specify both `atol` and `rtol`. Seems a bit involved to average users.

Reply:
@leofang Thanks for the Julia reference. Will dig a bit more on their issue threads to better understand their intent. We could be more specific here in naming `tol` -> `rtol`, which would pave the way for future support of `atol`.

Reply:
Re: complex. Yes, the machine eps should be the same as their corresponding real dtypes.